starboard/android/shared/media_codec_bridge.cc

```diff
@@ -180,7 +180,18 @@ std::unique_ptr<MediaCodecBridge> MediaCodecBridge::CreateAudioMediaCodecBridge(
 
   if (MediaCodecBridgeEradicator::GetInstance()->IsEnabled()) {
     // block if the old MediaCodecBridge instances haven't been destroyed yet
-    MediaCodecBridgeEradicator::GetInstance()->WaitForPendingDestructions();
+    bool signaled =
+        MediaCodecBridgeEradicator::GetInstance()->WaitForPendingDestructions();
+    if (!signaled) {
+      // timed out
+      std::string diagnostic_info_in_str = FormatString(
+          "MediaCodec destruction timeout: %d seconds, potential thread "
+          "leakage happened. Type = %s",
+          MediaCodecBridgeEradicator::GetInstance()->GetTimeoutSeconds(),
+          "Audio");
+      handler->OnMediaCodecError(false, false, diagnostic_info_in_str);
```

Comment:
We should double check if this works.

Reply:
how do I check it?

Reply:
Comment out the line "bool ...", and set signaled to false, and see if this error gets propagated to HTMLMediaElement.error.

Reply:
Tested, handler->OnMediaCodecError() can report error.